proto/viam/app/v1/app.proto

```diff
@@ -912,6 +912,15 @@ message RegistryItem {
     ModuleMetadata module_metadata = 11;
     MLModelMetadata ml_model_metadata = 12;
   }
```

Comment:
we will likely different types of metadata as we support more types in the registry, it's a breaking change either way but how would we do this if `metadata` is in the middle of the struct? not sure what the pb best practices are here

Reply:
Its totally acceptable to have protobufs where their fields are out of order when numbered. In the future it will probably look like this, and will work:

  oneof metadata {
    ModuleMetadata module_metadata = 11;
    MLModelMetadata ml_model_metadata = 12;
    SlamMapMetadata slam_metadata = 17;
  }

Remember that these fields are just use in serialize/deserialize of the message. In transit the payload is sent like `1=value,2=value` which allows the contents of the message to be shorter by not using field names. Also field names can change without breaking.

Anyways, I think `oneof` will be fine. The message with get tranferred with either 11, 12, or 17. and the gRPC client code will decipher that and make sure one is returned.

Also not sure we have other options than make the additional fields 50,51,52,53. Because we don't know how many types will be added over the years.

Reply:
Don't we preserve the serialization numbers (i.e. keeping metadata in the middle of the struct) in order to ensure backwards compatibility?

Reply:
> Not 100% sure: Don't we preserve the serialization numbers (i.e. keeping metadata in the middle of the struct) in order to ensure backwards compatibility?

Yeah we should for sure not renumber the metadata numbers as that will break clients.

Reply:
ah didn't realize you could do this! this sounds good to me then. an optional nit would be to put the new usage fields directly under the previous usage fields just so they're all in one place
```
...
int64 total_robot_usage = 9;
// The total number of organizations using this item
int64 total_organization_usage = 10;
// The total number of robots using this item outside of the owning org
int64 total_external_robot_usage = 13;
// The total number of organizations using this item outside of the owning org
int64 total_external_organization_usage = 14;

// Type-specific metadata
oneof metadata {
  ModuleMetadata module_metadata = 11;
  MLModelMetadata ml_model_metadata = 12;
}
```

Reply:
SGTM! will do that.